instrumentation/packaging/fpm/common.sh

```diff
@@ -57,20 +57,7 @@ download_java_agent() {
     local api_url=""
     local dl_url=""
 
-    if [ "$tag" = "latest" ]; then
-        tag=$( curl -sL "$JAVA_AGENT_RELEASE_URL/latest" | jq -r '.tag_name' )
-        if [ -z "$tag" ]; then
-            echo "Failed to get tag_name for latest release from $JAVA_AGENT_RELEASE_URL/latest" >&2
-            exit 1
-        fi
-    fi
-
-    api_url="$JAVA_AGENT_RELEASE_URL/tags/$tag"
-    dl_url="$( curl -sL "$api_url" | jq -r '.assets[] .browser_download_url' | grep "splunk-otel-javaagent.jar" | grep -v ".asc$" )"
-    if [ -z "$dl_url" ]; then
-        echo "Failed to get the agent download url from $api_url" >&2
-        exit 1
-    fi
+    dl_url="$JAVA_AGENT_RELEASE_URL/$tag/splunk-otel-javaagent.jar"
```

Comment:
This may prevent `latest` tags from working. Do you know if that's a problem for our pipeline and maybe we should keep the tag fetching api call?

```
curl https://github.com/signalfx/splunk-otel-java/releases/download/latest/splunk-otel-javaagent.jar
Not Found%
```

Reply:
No, latest with github resolves to the latest release. See https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases 
```
To link directly to a download of your latest release asset that was manually uploaded, link to /owner/name/releases/latest/download/asset-name.zip.
```

Reply:
So the `releases/download/$tag/<artifact>` release url should be updated to `releases/latest/download/<artifact>` if the tag is "latest"?

Reply:
Hmm, I'll review. I thought the build would pick this up.